NewForce-Cohort-7/tabloidfullstack-numpty-buggers
# Edit Post

Description
This pull request is for TabloidFullStack Trello ticket #8 https://trello.com/c/QPpd8KZH/8-8-edit-a-post Edit a post
When a user clicks on details of a post they can edit a post. They will click the edit button, be directed to a form filled with the information, then they will sumbit any changes made

The branch for the pull request: sl-postEdit

Updates

Updated PostRepository.cs, IPostRepository.cs, PostController.cs (C# Files)
Updated PostDetails.js, (React.js File)
Added PostEdit.js
Updated ApplicationViews.js to contain the path /posts/add
Type of change
New Feature - non-breaking

Testing Instructions
Follow these instructions to protect the branch you are working on and to test this pull request: https://gist.github.com/jordan-castelloe/96f76d5bd4cfbd2181084af674c2a1af#when-you-need-to-test-a-teammates-pr

Seed The Data
If you do not currently have the code for this project, clone the repository - (https://github.com/NewForce-Cohort-7/tabloidfullstack-numpty-buggers)

Open the TabloidFullStack.sln in visual studio and prepare your database.
In the github repository, open the SQL folder: https://github.com/NewForce-Cohort-7/tabloidfullstack-numpty-buggers/tree/main/SQL)
there are two files - [01_Tabloid_Create_DB.sql](https://github.com/NewForce-Cohort-7/tabloidfullstack-numpty-buggers/blob/main/SQL/01_Tabloid_Create_DB.sql), [02_Tabloid_Seed_Data.sql](https://github.com/NewForce-Cohort-7/tabloidfullstack-numpty-buggers/blob/main/SQL/02_Tabloid_Seed_Data.sql)

In Visual Studio in the top navbar click View and select SQL Server Object Explorer. Open the server marked as "SQLEXPRESS" in the SQL Server Object Explorer on the left side of Visual Code. Once there, add a new SQL server by selecting the 'Add SQL Server' option at the top of the side bar. Copy the data from the seed data and execute by pressing the green Play button.
start the server by clicking the solid green play button in the middle top of your screen. A browser window will display with the Tabloid landing page.

Test the edit post feature

Once in the branch sl-postEdit and having booted up VS and performed "npm start" at the tabloid folder you should have swagger and localhost:3000 pulled up, login as [[[foo@bar.com](mailto:foo@bar.com)] (no password needed)

1. Click on "Posts" in the navbar and you should see a list of all the posts.
2. Click on the "Post Details" linkand you will directed to post details
3. Click the edit button
4. Fill out the form as desired
5. Click submit
6. You can click details of that post and see the corrections you have made
